sky/jobs/recovery_strategy.py

```diff
@@ -289,26 +301,53 @@ def _launch(self,
                         state.set_restarting(self.job_id, self.task_id,
                                              recovery)
                     try:
-                        usage_lib.messages.usage.set_internal()
-                        # Detach setup, so that the setup failure can be
-                        # detected by the controller process (job_status ->
-                        # FAILED_SETUP).
-                        execution.launch(
-                            self.dag,
-                            cluster_name=self.cluster_name,
-                            # We expect to tear down the cluster as soon as the
-                            # job is finished. However, in case the controller
-                            # dies, set autodown to try and avoid a resource
-                            # leak.
-                            idle_minutes_to_autostop=_AUTODOWN_MINUTES,
-                            down=True,
-                            _is_launched_by_jobs_controller=True)
-                        logger.info('Managed job cluster launched.')
+                        if not state.force_no_postgres():
```

Comment:
Why are we using this for the if condition?? What does it have to do with postgres? Maybe just need to rename this function, but it doesn't make sense right now.